glean-core/src/lib.rs

```diff
@@ -229,6 +229,25 @@ fn setup_state(state: State) {
     }
 }
 
+/// A global singleton that stores listener callbacks registered with Glean
+/// to receive event recording notifications.
+static EVENT_LISTENERS: OnceCell<Mutex<Vec<Box<dyn GleanEventListener>>>> = OnceCell::new();
+
+fn event_listeners() -> &'static Mutex<Vec<Box<dyn GleanEventListener>>> {
+    if EVENT_LISTENERS.get().is_none() && EVENT_LISTENERS.set(Mutex::new(vec![])).is_err() {
+        log::error!(
+            "Global Glean event listeners object is initialized already. This probably happened concurrently."
+        )
+    }
+
+    return EVENT_LISTENERS.get().unwrap();
```

Comment:
then this becomes just:

```suggestion
    return &EVENT_LISTENERS;
```

Reply:
Nice! I was imitating the other callbacks, good to know this can be simplified. :)

Reply:
yeah, it's const since Rust 1.63: https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.new
We might update the others in a followup, though we should check that it's actually the same behavior.